File: src/pi.rb
puts "Hello World, From Ruby!"
NUMBER_OF_ITERATIONS = 3000
NUMBER_OF_ITERATIONS_FOR_AVG_TIME = 100

def main
    iteration = 1
    total_time = 0.0
    until iteration > NUMBER_OF_ITERATIONS_FOR_AVG_TIME
        start = Process.clock_gettime(Process::CLOCK_MONOTONIC)
        calculate_pi_with_wallis_product(NUMBER_OF_ITERATIONS)
        stop = Process.clock_gettime(Process::CLOCK_MONOTONIC)
        total_time = total_time + (stop - start)
        iteration = iteration + 1
    end
    avg_time_taken = total_time.fdiv(NUMBER_OF_ITERATIONS_FOR_AVG_TIME)
    result = calculate_pi_with_wallis_product(NUMBER_OF_ITERATIONS)
    puts "Value of pi calcualted iteratively with the Madhava-Leibniz series: " + result.to_s + ". \nNumber of Iterations: " + NUMBER_OF_ITERATIONS.to_s + ". \nAverage Time Taken: " + (avg_time_taken * 1000).to_s + "(in ms)."
end

def calculate_pi_with_wallis_product iterations
    product = 1.0
    current_iteration = 1
    until current_iteration > iterations
        top = 4.0 * current_iteration * current_iteration
        bottom = 4.0 * current_iteration * current_iteration - 1.0
        product = product * top.fdiv(bottom)
        current_iteration = current_iteration + 1
    end
    return product * 2
end

main

File: src/pi.lua
print("Hello, World From Lua!")
function main()
    NUM_ITERATIONS = 1000
    NUM_ITERATIONS_FOR_TIME_AVG = 100
    time_sum = 0   
    result = pi_from_madhava_leibniz_series(NUM_ITERATIONS)
    print(string.format("Value of pi calcualted iteratively with the Madhava-Leibniz series: %f. \nNumber of Iterations: %d. \nNo time measured because lua os.clock() is system dependent... may god save our souls.", result, NUM_ITERATIONS))
end

function pi_from_madhava_leibniz_series(num_iterations)
    sum = 0
    for n = 0, num_iterations-1, 1 do
        sum = sum + (n%2 == 1 and -1 or 1) / (2*n + 1)
    end
    return sum * 4
end

main()